pkg/test/env/factory_wiremock.go

```diff
@@ -0,0 +1,85 @@
+package env
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/mon"
+	"io/ioutil"
+	"net/http"
+)
+
+func init() {
+	componentFactories[componentWiremock] = new(wiremockFactory)
+}
+
+const componentWiremock = "wiremock"
+
+type wiremockSettings struct {
+	ComponentBaseSettings
+	ComponentContainerSettings
+	Mocks string `cfg:"mocks"`
+	Port  int    `cfg:"port" default:"0"`
+}
+
+type wiremockFactory struct {
+}
+
+func (f *wiremockFactory) Detect(_ cfg.Config, _ *ComponentsConfigManager) error {
+	return nil
+}
+
+func (f *wiremockFactory) GetSettingsSchema() ComponentBaseSettingsAware {
+	return &wiremockSettings{}
+}
+
+func (f *wiremockFactory) ConfigureContainer(settings interface{}) *containerConfig {
+	s := settings.(*wiremockSettings)
+
+	return &containerConfig{
+		Repository: "rodolpheche/wiremock",
+		Tag:        "2.26.3-alpine",
+		PortBindings: portBindings{
+			"8080/tcp": s.Port,
+		},
+		ExpireAfter: s.ExpireAfter,
+	}
+}
+
+func (f *wiremockFactory) HealthCheck(_ interface{}) ComponentHealthCheck {
+	return func(container *container) error {
+		binding := container.bindings["8080/tcp"]
+		url := f.getUrl(binding)
+
+		_, err := http.Get(url)
+
+		return err
+	}
+}
+
+func (f *wiremockFactory) Component(_ cfg.Config, logger mon.Logger, container *container, settings interface{}) (Component, error) {
+	component := &wiremockComponent{
+		logger:  logger,
+		binding: container.bindings["8080/tcp"],
+	}
+
+	s := settings.(*wiremockSettings)
+	jsonStr, err := ioutil.ReadFile(s.Mocks)
+
+	if err != nil {
+		return nil, fmt.Errorf("could not read http mock configuration: %w", err)
+	}
+
+	url := f.getUrl(container.bindings["8080/tcp"])
+	_, err = http.Post(url+"/mappings/import", "application/json", bytes.NewBuffer(jsonStr))
```

Comment:
What happens if that request returns an error response?

Reply:
The whole component will not come up and due to that will error out here: pkg/test/mocks.go:56

Reply:
I don't see how that is related. When testing it locally, my wiremock stayed healthy even though the import failed with status 422 (I did sent invalid JSON).